File: favourite-lyrics-jpa/src/main/java/org/jesperancinha/lyrics/jpa/model/LyricsEntity.java
package org.jesperancinha.lyrics.jpa.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.Hibernate;

import java.util.Objects;
import java.util.UUID;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "lyrics")
@Getter
@Setter
@ToString
public class LyricsEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column
    private UUID id;

    @Column
    private String lyrics;

    @Column
    private String participatingArtist;

    @Override
    public boolean equals(Object o) {
        if(o==null){
            return false;
        }
        if(!(o instanceof LyricsEntity that)){
            return false;
        }
        if (this == o) return true;
        if (Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        return id != null && Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: favourite-lyrics-jpa/src/main/kotlin/org/jesperancinha/lyrics/jpa/model/LyricsEntity.kt
package org.jesperancinha.lyrics.jpa.model

import jakarta.persistence.*
import org.hibernate.Hibernate
import java.util.*

@Entity
@Table(name = "lyrics")
data class LyricsEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column
    private val id: UUID? = null,

    @Column
    private val lyrics: String? = null,

    @Column
    private val participatingArtist: String? = null,
) {

    override fun equals(o: Any?): Boolean {
        if (o == null) {
            return false
        }
        if (o !is LyricsEntity) {
            return false
        }
        if (this === o) return true
        return if (Hibernate.getClass(this) != Hibernate.getClass<Any>(o)) false else id != null && id == o.id
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}